### PR TITLE
fix: network add form

### LINKED ISSUE #5401: Fix regtest addresses
@Polybius93 noticed addresses aren't correct on regtest.

> Also, there is some problem with the regtest network support in Leather, whenever I add a new regtest network, the addresses are ``bc1…`` addresses, instead of ``bcrt1…`` so it is not working properly.

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build 0d0468b — [Extension build](https://github.com/leather-wallet/extension/actions/runs/9176128349), [Test report](https://leather-wallet.github.io/playwright-reports/fix/regtest-address), [Storybook](https://fix-regtest-address--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=fix/regtest-address)

This pr fixes bitcoin network api selector, regtest network address generation, refactors network form to use our new `Select` component, also contains small ui fix for select component. 

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `Select` component for improved network selection.

- **Enhancements**
  - Updated the `useAddNetwork` function to include `bitcoinApi` field for better network configuration.

- **UI Improvements**
  - Enhanced the `Item` component within the `Select` for a more consistent styling experience.

### CODE REVIEW

**Comment by fbwoolf** [comment]
@alter-eggo is this related to an issue that can be linked here?

**Comment by kyranjamie** [comment]
@alter-eggo does this fix the issue where Regtest addresses aren't generated with the correct prefix?

**Comment by alter-eggo** [comment]
@kyranjamie yes, btc address starts from `bcrt1`
Screenshot 2024-05-21 at 18 02 12

File: src/app/features/add-network/use-add-network.tsx [hunk 1]
```diff
@@ -27,31 +27,30 @@ export interface AddNetworkFormValues {
   name: string;
   stacksUrl: string;
   bitcoinUrl: string;
+  bitcoinApi: BitcoinNetworkModes;
```
Comment (kyranjamie) [inline src/app/features/add-network/use-add-network.tsx]: `bitcoinNetwork`?

**Review by kyranjamie — APPROVED** [review]
Naming comment, otherwise 👍🏼

File: src/app/features/add-network/add-network-form.tsx [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
 import { Input } from '@app/ui/components/input/input';
+import { Select } from '@app/ui/components/select/select';
+import { SelectItemLayout } from '@app/ui/components/select/select-item.layout';
 import { Title } from '@app/ui/components/typography/title';
+import { CheckmarkIcon, ChevronDownIcon } from '@app/ui/icons';
 
-import { type AddNetworkFormValues, useAddNetwork } from './use-add-network';
+import { type AddNetworkFormValues } from './use-add-network';
+
+const networks = [
```
Comment (kyranjamie) [inline src/app/features/add-network/add-network-form.tsx]: Can we strongly type, so `BitcoinNetworkMode` is used?

**Review by fbwoolf — APPROVED** [review]
Good to see the Select component being used. 👍

**Pull request merged**